test/index.test.js

```diff
@@ -57,4 +59,306 @@ describe('react-ga', function() {
+      getGaCalls().should.eql([ [ 'create', 'foo', 'auto' ],
+                                [ 'send', { eventAction: 'Send Test',
+                                            eventCategory: 'Test',
+                                            nonInteraction: true,
+                                            hitType: 'event'
+                                          }]
+                                ]);
+    });
+
+    it('should reject a non-boolean nonInteraction value', function() {
+      ga.initialize('foo');
+      ga.event( { category: 'Test', action: 'Send Test', nonInteraction: 'yeahsure' } );
+      console.warn.args.should.eql([[
+        '[react-ga]', '`args.nonInteraction` must be a boolean.'
+      ]]);
+      getGaCalls().should.eql([ [ 'create', 'foo', 'auto' ],
+                                [ 'send', { eventAction: 'Send Test',
+                                            eventCategory: 'Test',
+                                            hitType: 'event'
+                                          }]
+                                ]);
+    });
+  });
+
+  /**
+   * outboundLink()
+   */
+
+  describe('outboundLink()', function() {
+
+    it('should record an outboundLink event', function() {
+
+      ga.initialize('foo');
+      ga.outboundLink( { label: 'Test Click' }, function () {
+      } );
+
+      getGaCalls().should.eql([ [ 'create', 'foo', 'auto' ],
+                                [ 'send', { eventAction: 'Click',
+                                            eventCategory: 'Outbound',
+                                            eventLabel: 'Test Click',
+                                            hitCallback: undefined,
+                                            hitType: 'event'
+                                          }]
+                                ]);
+
+    });
+
+    it('should warn if all args are missing', function() {
+      ga.initialize('foo');
+      ga.outboundLink();
+      console.warn.args.should.eql([[
+        '[react-ga]', 'hitCallback function is required'
+      ]]);
+    });
+
+    it('should warn if label arg is missing', function() {
+      ga.initialize('foo');
+      ga.outboundLink( { missing: 'labelarg' }, function () {
+        // do nothing on callback
+      } );
+      console.warn.args.should.eql([[
+        '[react-ga]', 'args.label is required in outboundLink()'
+      ]]);
+    });
+
+    it('should warn if hitCallback is missing', function() {
+      ga.initialize('foo');
+      ga.outboundLink( { label: 'Missing hitCallback' } );
+      console.warn.args.should.eql([[
+        '[react-ga]', 'hitCallback function is required'
+      ]]);
+    });
+
+    it('should fire hitCallback even if ga is not defined', function(done) {
+      ga.outboundLink( { label: 'http://www.google.com' }, function () {
+        done();
+      });
+    });
+
+    // TODO: solve how to simulate lack of response from GA
```

Comment:
Oh, since we're already able to access a list of all the calls to `ga()` I would recommend getting the reference to `fieldObject` that you've passed in at [that point](https://github.com/adamlofting/react-ga/pull/13/files#diff-1fdf421c05c1140f6d71444ea2b27638R271) from the return value of `getGaCalls()`, and then calling `fieldObject.hitCallback()` from the test to "simulate" the hit callback being called by the GA code (or not called, as the case may be). 

You'd probably want to use sinon's fake timers to simulate all the different ways things could happen, e.g.:

* The hit callback is never called back,
* The hit callback is called back within 250ms,
* The hit callback is called after 250ms (i.e., you want to make sure you don't call the passed-in callback *twice*).

Does that make sense? To clarify, I think that the two main things you need are:

1. Getting a reference to the hit callback that you've passed `ga()`, so you can "fake" the Google Analytics code calling it.
2. Using sinon's fake timers to simulate the passage of time.